### PR TITLE
BLD: Adjust Numpy Minimum Versions for aarch64/arm64 compatibility

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,9 +5,11 @@ requires = [
...
     "wheel",
     "Cython>=0.29.21,<3",  # Note: sync with setup.py
-    "numpy==1.17.3; python_version=='3.7'",
-    "numpy==1.18.3; python_version=='3.8'",
-    "numpy; python_version>='3.9'",
+    # Note: ARM64 needs newer numpy
+    "numpy==1.17.3; python_version=='3.7' and platform_machine!='arm64'",
+    "numpy==1.18.3; python_version=='3.8' and platform_machine!='arm64'",
+    "numpy>=1.19.3; python_version>='3.9' and platform_machine!='arm64'",
+    "numpy>=1.20.0; platform_machine=='arm64'"
```
Comment (fangchenli) [inline pyproject.toml:12]: See https://github.com/scipy/scipy/blob/master/pyproject.toml. We might need something like that.

**Comment by fangchenli** [comment]
We need to backport this. Currently, `pip install pandas --pre` doesn't work on M1 mac.

**Comment by jreback** [comment]
ok let me look 

we should merge the sdist test first

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @lithomas1

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Comment by aptalca** [comment]
Hi,
It looks like this PR incorrectly excluded all python 3.9 numpy version requirements on aarch64 and since the backports, I can't build any pandas 1.3.X versions on aarch64 with python 3.9.

Opened a PR to fix it: #43722 just fyi

Thanks

**Pull request merged**